guide/src/class.md

```diff
@@ -434,10 +431,9 @@ impl DictWithCounter {
         Self::default()
     }
 
-    fn set(mut self_: PyRefMut<'_, Self>, key: String, value: &PyAny) -> PyResult<()> {
-        self_.counter.entry(key.clone()).or_insert(0);
-        let py = self_.py();
-        let dict: &PyDict = unsafe { py.from_borrowed_ptr_or_err(self_.as_ptr())? };
+    fn set(slf: &Bound<'_, Self>, key: String, value: &PyAny) -> PyResult<()> {
+        slf.borrow_mut().counter.entry(key.clone()).or_insert(0);
+        let dict = slf.downcast::<PyDict>()?;
```

Comment:
👍 yep I think even though `downcast_unchecked` would be ok here I think best to recommend the safe option in the guide.

Reply:
Yes, that was the intention. The code from the guide is often copied as a first step, so `unsafe` should be avoided I think.